CrabZoomer.js

```diff
@@ -141,141 +143,153 @@ define([
     },
 
     _municipalityChange: function () {
       //console.debug('CrabZoomer::_municipalityChange');
       var value = domUtils.getSelectedOption(this.municipalitySelect);
 
-      this._setStreet('');
-      this._setNumber('');
       this.disable();
+      this._fillStreetSelect([]);
+      this._fillNumberSelect([]);      
 
       if (!value) {
-        this._fillStreetSelect([]);
-        this._fillNumberSelect([]);
-        domAttr.remove(this.provinceSelect, "disabled");
-        domAttr.remove(this.municipalitySelect, "disabled");
+        if (this.alleGewesten) {
+          domAttr.remove(this.gewestSelect, 'disabled');
+        }
+        if (this.provinceList.length > 0){
+          domAttr.remove(this.provinceSelect, 'disabled');
+        }
+        domAttr.remove(this.municipalitySelect, 'disabled');
         return false;
       }
 
-      var self = this;
-
-      request(this.baseUrl + "/crab/gemeenten/" + value + "/straten?aantal=5000", {
-        handleAs: "json",
-        headers: {
-          "X-Requested-With": ""
-        }
-      }).then(
-        function (jsondata) {
-          if (self.sortMethod) {
-            jsondata.sort(self.sortMethod);
+      this._crabGet('gemeenten/' + value + '/straten', this.sortMethod).then(
+        lang.hitch(this, function (jsondata) {
+          this._fillStreetSelect(jsondata);
+          if (this.alleGewesten) {
+            domAttr.remove(this.gewestSelect, 'disabled');
           }
+          if (this.provinceList.length > 0){
+            domAttr.remove(this.provinceSelect, 'disabled');
+          }
+          domAttr.remove(this.municipalitySelect, 'disabled');
+          domAttr.remove(this.streetSelect, 'disabled');
 
-          self._fillStreetSelect(jsondata);
-
-          domAttr.remove(self.provinceSelect, "disabled");
-          domAttr.remove(self.municipalitySelect, "disabled");
-          domAttr.remove(self.streetSelect, "disabled");
-
-          var location = self.value;
-          if (location && location.street && location.municipality && location.municipality.id == value) {
-            self._setStreet(location.street.id);
+          var location = this.value;
+          if (location && location.street && location.municipality && location.municipality.id === value) {
+            this._setStreet(location.street.id);
           }
-        },
-        function (error) {
-          self._errorHandler(error);
-        }
+        })
       );
     },
 
     _streetChange: function () {
       //console.debug('CrabZoomer::_streetChange');
       var value = domUtils.getSelectedOption(this.streetSelect);
 
-      this._setNumber('');
       this.disable();
+      this._fillNumberSelect([]); 
 
       if (!value) {
-        this._fillNumberSelect([]);
-        domAttr.remove(this.provinceSelect, "disabled");
-        domAttr.remove(this.municipalitySelect, "disabled");
-        domAttr.remove(this.streetSelect, "disabled");
+        if (this.alleGewesten) {
```

Comment:
Ik denk dat ge deze blok waarbij ge die disableds weghaalt in een functie kunt steken en geleidelijk aan kan afchecken gebaseerd op je type (gewest/provincie/gemeente/straat). Dan gaat het misschien overzichtelijker worden?

Reply:
true that, maar deze component is deprecated (dojo) en 
![image](https://user-images.githubusercontent.com/1849458/109967695-06073e80-7cf2-11eb-966b-10fe51732dae.png)

Annelies wil straks testen en vant weekend in productie gaan.